### ISSUE TITLE
getAttributeDefinitions will generate duplicate AttributeDefinitions

### ISSUE THREAD

**Issue by exalted** [architect/architect#828]
**The issue**

Given the following partial `.arc` file:

```
@tables
data
  PK *String # Partition key
  SK **String # Sort key

@indexes
data
  PK *String
  foo **String # Another attribute to "rehash" items in PK collection
```

[getAttributeDefinitions function](https://github.com/architect/sandbox/blob/37cabab26518a6f990aa1aa08766212cad9f2291/src/db/create-table/_get-attribute-definitions-with-gsi.js#L16) will generate duplicate attribute definitions like this:

```
AttributeDefinitions: [
  { AttributeName: 'PK', AttributeType: 'S' },
  { AttributeName: 'foo', AttributeType: 'S' },
  { AttributeName: 'PK', AttributeType: 'S' },  # <-- duplicate !!!
  { AttributeName: 'SK', AttributeType: 'S' }
]
```

This doesn't seem to be an issue with `dynalite` (I didn't verify why, but I think it's possibly due to lousy validation checks in the library). However if you wanted to connect to a [DynamoDB Local](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.html) server running as such: `dynamodb-local -port 5000 & arc sandbox` you'll get the following error message: `ValidationException: Cannot have two attributes with the same name`.

**Expected behavior**

I expect the table and the GSI to be created successfully regardless of the fact that the server is dynalite or DynamoDB Local.

**Versions**

 - OS: macOS Catalina 10.15.4 (19E287)
- Architect: 6.4.0

**Comment by filmaj** [comment]
Yep, can confirm this issue.

I think we can resolve this by having `getAttributeDefinitionsWithGSI` de-dupe the compiled final table attributes from across both the base table definition as well as all the indices.

I'll put together a PR for this.

**Comment by exalted** [comment]
> I think we can resolve this by having `getAttributeDefinitionsWithGSI` de-dupe the compiled final table attributes from across both the base table definition as well as all the indices.

Sounds good to me. 👍

Thank you!

**Comment by ryanblock** [comment]
Should be fixed in `1.12.3` courtesy of @filmaj!
